src/DynamoCore/Graph/Workspaces/WorkspaceInfo.cs

```diff
@@ -184,7 +186,8 @@ public WorkspaceInfo()
                 // we have a dyf and it lacks an ID field, we need to assign it
                 // a deterministic guid based on its name.  By doing it deterministically,
                 // files remain compatible
-                if (string.IsNullOrEmpty(id) && !string.IsNullOrEmpty(funName) && funName != Properties.Resources.DefaultHomeWorkspaceName)
+                //TODO(mjk) we should get rid of this and throw instead since non hame names are now valid in json format.
+                if (string.IsNullOrEmpty(id) && !string.IsNullOrEmpty(funName) && funName != dynamo1HomeWorkspaceNameString)
```

Comment:
@QilongTang do you remember why you added this? I guess at the time you added it here for JSON deserialization it was still valid, but now that json home workspace names can be other strings besides "Home" I think we should use the cusotmNode flag here instead or just throw if the UUID is null.